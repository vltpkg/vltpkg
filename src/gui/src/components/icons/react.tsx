export const React = (props: React.SVGProps<SVGSVGElement>) => {
  return (
    <svg
      width="500"
      height="500"
      viewBox="0 0 500 500"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}>
      <g clip-path="url(#clip0_2967_809)">
        <path
          d="m250.504 284.862c270.046 284.862 285.889 269.029 285.889 249.499c285.889 229.968 270.046 214.136 250.504 214.136c230.961 214.136 215.119 229.968 215.119 249.499c215.119 269.029 230.961 284.862 250.504 284.862z"
          fill="currentColor"
        />
        <path
          d="m250.501 321.96c355.363 321.96 440.371 289.522 440.371 249.509c440.371 209.495 355.363 177.058 250.501 177.058c145.639 177.058 60.6318 209.495 60.6318 249.509c60.6318 289.522 145.639 321.96 250.501 321.96z"
          stroke="currentColor"
          stroke-width="18.913"
        />
        <path
          d="m187.724 285.725c240.155 376.482 310.768 433.837 345.442 413.83c380.116 393.823 365.721 304.031 313.29 213.274c260.859 122.517 190.247 65.1621 155.572 85.1689c120.898 105.176 135.293 194.968 187.724 285.725z"
          stroke="currentColor"
          stroke-width="18.913"
        />
        <path
          d="m187.726 213.277c135.295 304.034 120.9 393.826 155.574 413.833c190.248 433.84 260.861 376.485 313.292 285.728c365.723 194.971 380.118 105.179 345.443 85.172c310.769 65.1652 240.157 122.52 187.726 213.277z"
          stroke="currentColor"
          stroke-width="18.913"
        />
      </g>
      <defs>
        <clipPath id="clip0_2967_809">
          <rect
            width="397"
            height="353"
            fill="currentColor"
            transform="translate(52 73)"
          />
        </clipPath>
      </defs>
    </svg>
  )
}
