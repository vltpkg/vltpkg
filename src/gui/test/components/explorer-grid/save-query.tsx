import React from 'react'
import { vi, describe, it, expect, afterEach } from 'vitest'
import { cleanup, render } from '@testing-library/react'
import html from 'diffable-html'
import { useGraphStore as useStore } from '@/state/index.js'
import SaveQueryButton from '@/components/explorer-grid/save-query.jsx'

vi.mock('@/components/ui/popover.jsx', () => ({
  Popover: 'gui-popover',
  PopoverContent: 'gui-popover-content',
  PopoverTrigger: 'gui-popover-trigger',
}))

vi.mock('@/components/ui/tooltip.jsx', () => ({
  Tooltip: 'gui-tooltip',
  TooltipContent: 'gui-tooltip-content',
  TooltipProvider: 'gui-tooltip-provider',
  TooltipTrigger: 'gui-tooltip-trigger',
}))

vi.mock('@/components/ui/button.jsx', () => ({
  Button: 'gui-button',
}))

vi.mock('@/components/ui/label.jsx', () => ({
  Label: 'gui-label',
}))

vi.mock('@/components/ui/input.jsx', () => ({
  Input: 'gui-input',
}))

vi.mock('@/components/labels/label-select.jsx', () => ({
  LabelSelect: 'gui-label-select',
}))

vi.mock('@/components/labels/label-badge.jsx', () => ({
  LabelBadge: 'gui-label-badge',
}))

vi.mock('@/components/queries/delete-query.jsx', () => ({
  DeleteQuery: 'gui-delete-query',
}))

expect.addSnapshotSerializer({
  serialize: v => html(v),
  test: () => true,
})

afterEach(() => {
  const CleanUp = () => (useStore(state => state.reset)(), '')
  render(<CleanUp />)
  cleanup()
})

describe('save-query', () => {
  it('should render correctly', () => {
    const Container = () => {
      return <SaveQueryButton />
    }

    const { container } = render(<Container />)

    expect(container.innerHTML).toMatchSnapshot()
  })
})
