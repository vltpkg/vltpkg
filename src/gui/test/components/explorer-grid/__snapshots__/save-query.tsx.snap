// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`save-query > should render correctly 1`] = `

<gui-popover open="false">
  <gui-popover-trigger classname="flex items-center justify-center">
    <gui-tooltip-provider>
      <gui-tooltip>
        <gui-tooltip-trigger
          aschild="true"
          classname="flex h-[1.5rem] w-[1.5rem] items-center justify-center rounded-sm border border-muted-foreground/20 bg-muted"
        >
          <div class="inline-flex h-[1.5rem] w-[1.5rem] items-center justify-center gap-2 whitespace-nowrap rounded-md border border-input bg-background text-sm font-medium ring-offset-background transition-colors hover:bg-accent hover:text-accent-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&amp;_svg]:pointer-events-none [&amp;_svg]:size-4 [&amp;_svg]:shrink-0">
            <svg
              xmlns="http://www.w3.org/2000/svg"
              width="20"
              height="20"
              viewbox="0 0 24 24"
              stroke="currentColor"
              stroke-width="2"
              stroke-linecap="round"
              stroke-linejoin="round"
              class="lucide lucide-star"
              fill="#fafafa"
            >
              <polygon points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2">
              </polygon>
            </svg>
          </div>
        </gui-tooltip-trigger>
        <gui-tooltip-content>
          <p>
            Save query
          </p>
        </gui-tooltip-content>
      </gui-tooltip>
    </gui-tooltip-provider>
  </gui-popover-trigger>
  <gui-popover-content
    classname="w-full p-0"
    align="end"
  >
  </gui-popover-content>
</gui-popover>

`;
